src/UserService.Business/Commands/Gender/FindGenderCommand.cs

```diff
@@ -0,0 +1,45 @@
+﻿using LT.DigitalOffice.Kernel.Enums;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.UserService.Business.Commands.Gender.Interfaces;
+using LT.DigitalOffice.UserService.Data.Interfaces;
+using LT.DigitalOffice.UserService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.UserService.Models.Db;
+using LT.DigitalOffice.UserService.Models.Dto.Models;
+using LT.DigitalOffice.UserService.Models.Dto.Requests.Gender.Filters;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.UserService.Business.Commands.Gender
+{
+  public class FindGenderCommand : IFindGenderCommand
+  {
+    private readonly IGenderRepository _genderRepository;
+    private readonly IGenderInfoMapper _mapper;
+
+    public FindGenderCommand(
+      IGenderRepository genderRepository,
+      IGenderInfoMapper mapper)
+    {
+      _genderRepository = genderRepository;
+      _mapper = mapper;
+    }
+    public async Task<FindResultResponse<GenderInfo>> ExecuteAsync(FindGendersFilter filter)
```

Comment:
you get exception if takeCount <= 0, add validation

Reply:
fixed